internal/plugin/connectors/http/generic/oauth/v1/protocol_test.go

```diff
@@ -0,0 +1,352 @@
+		name string
+		args args
+		want int
+	}{
+		{
+			name: "Generates a 16 length nonce",
+			args: args{
+				length: 16,
+			},
+			want: 16,
+		},
+		{
+			name: "Generates a 32 length nonce",
+			args: args{
+				length: 32,
+			},
+			want: 32,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := len(generateNonce(tt.args.length, nonceCharset)); got != tt.want {
+				t.Errorf("generateNonce() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_generateSigningKey(t *testing.T) {
+	type args struct {
+		oauth1 oAuth1
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    string
+		wantErr bool
+	}{
+		{
+			name: "Generates a valid signing key: 2 keys",
+			args: args{
+				oAuth1{
+					ConsumerSecret: "kAcSOqF21Fu85e7zjz7ZN2U4ZRhfV3WpwPAoE3Z7kBw",
+					TokenSecret:    "LswwdoUaIvS8ltyTt5jkRh4J50vUPVVHtR2YPi5kE",
+				},
+			},
+			want: "kAcSOqF21Fu85e7zjz7ZN2U4ZRhfV3WpwPAoE3Z7kBw&LswwdoUaIvS8ltyTt5jkRh4J50vUPVVHtR2YPi5kE",
+		},
+		{
+			name: "Generates a valid signing key: No token secret",
+			args: args{
+				oAuth1{
+					ConsumerSecret: "kAcSOqF21Fu85e7zjz7ZN2U4ZRhfV3WpwPAoE3Z7kBw",
+					TokenSecret:    "",
+				},
+			},
+			want: "kAcSOqF21Fu85e7zjz7ZN2U4ZRhfV3WpwPAoE3Z7kBw&",
+		},
+		{
+			name: "Generates a valid signing key: Both empty",
+			args: args{
+				oAuth1{
+					ConsumerSecret: "",
+					TokenSecret:    "",
+				},
+			},
+			want: "&",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := generateSigningKey(tt.args.oauth1)
+			if got != tt.want {
+				t.Errorf("generateSigningKey() got = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_constructOAuthString(t *testing.T) {
+	oauth1 := oAuth1{
+		ConsumerKey:     "xvz1evFS4wEEPTGEFPHBog",
+		ConsumerSecret:  "kAcSOqF21Fu85e7zjz7ZN2U4ZRhfV3WpwPAoE3Z7kBw",
+		Nonce:           "kYjzVBB8Y0ZFabxSWbWovY3uYSQ2pTgmZeNu2VS4cg",
+		Signature:       "tnnArxj06cWHq44gCs1OSKk%2FjLY%3D",
+		SignatureMethod: "HMAC-SHA1",
+		TimeStamp:       "1318622958",
+		Token:           "370773112-GmHxMAgYyLbNEtIKZeRNFsMKPR9EyMZeS9weJAEb",
+		TokenSecret:     "LswwdoUaIvS8ltyTt5jkRh4J50vUPVVHtR2YPi5kE",
+		Version:         "1.0",
+	}
+
+	result, _ := constructOAuthString(oauth1)
+
+	want := "OAuth oauth_consumer_key=\"xvz1evFS4wEEPTGEFPHBog\", oauth_nonce=\"kYjzVBB8Y0ZFabxSWbWovY3uYSQ2pTgmZeNu2VS4cg\", oauth_signature=\"tnnArxj06cWHq44gCs1OSKk%2FjLY%3D\", oauth_signature_method=\"HMAC-SHA1\", oauth_timestamp=\"1318622958\", oauth_token=\"370773112-GmHxMAgYyLbNEtIKZeRNFsMKPR9EyMZeS9weJAEb\", oauth_version=\"1.0\""
+
+	if result == want {
+		t.Log("constructOAuthString() Passed")
+	} else {
+		t.Errorf("extractOAuthParams() FAILED, expected %v but got %v", want, result)
+	}
+}
+
+func Test_encodeURI(t *testing.T) {
+	type args struct {
+		stringToEncode string
+	}
+	tests := []struct {
+		name string
+		args args
+		want string
+	}{
+		{
+			name: "Validate encodeURI(): 1",
```

Comment:
So I have a few thoughts on this test:
1. Let's change the naming of these tests to describe the situations we're covering. The context of a test can be hard to determine from the values being passed in and out.
2. On [this page](https://tools.ietf.org/html/rfc5849#section-3.6) you shared, we can see a few cases we should test specifically.
- Characters in the unreserved character set as defined by
          [RFC3986], Section 2.3 (ALPHA, DIGIT, "-", ".", "_", "~") MUST
          NOT be encoded.
- All other characters MUST be encoded.
- The two hexadecimal characters used to represent encoded
          characters MUST be uppercase. 

Let's make sure we test these cases as well as add names that make it clear that we're testing 'em